Sources/Utility/POSIXConvenience.swift

```diff
@@ -11,11 +11,9 @@
 import POSIX
 
 public func fopen(path: String..., mode: FopenMode = .Read, body: (UnsafeMutablePointer<FILE>) throws -> Void) throws {
-    var fp = try POSIX.fopen(Path.join(path), mode: mode)
-    defer { if fp != nil { fclose(fp) } }
+    let fp = try POSIX.fopen(Path.join(path), mode: mode)
+    defer { fclose(fp) }
     try body(fp)
-    fclose(fp)  // defer is not necessarily immediate
```

Comment:
{{defer}} will always happen at the end of the scope (close-brace), just like a C++ destructor. It can get moved around by optimization, but so can everything else.